mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -74,4 +74,69 @@ export class WayPreviewDAL {
     return waysPreview;
   }
 
+  /**
+   * Get User own ways preview
+   */
+  public static async getOwnWaysPreview(uuid: string): Promise<WayPreview[]> {
+    const waysDTO = await WayService.getOwnWaysDTO(uuid);
+    const usersPreview = await UserPreviewDAL.getUsersPreview();
+    const goalsPreview = await GoalPreviewDAL.getGoalsPreview();
+
+    const ownersPreview = waysDTO.map((wayDTO) => {
+      const ownerPreview = usersPreview
+        .find((elem) => elem.uuid === wayDTO.ownerUuid);
+      if (!ownerPreview) {
+        throw new Error(`${ownerPreview} was not found`);
+      }
+
+      return ownerPreview;
+    });
+
+    /**
+     * Get currentMentors from currentMentorUuids for each way
+     */
+    const currentMentorsPreview = waysDTO.map((wayDTO) => {
+      const currentMentorPreview = wayDTO.currentMentorUuids.map((userUuid: string) => {
+        const mentorsPreview = usersPreview
+          //TODO: task #114 Use hashmap instead of .find
+          .find((elem) => elem.uuid === userUuid);
+        if (!mentorsPreview) {
+          throw new Error(`${mentorsPreview} was not found`);
+        }
+
+        return mentorsPreview;
+      });
+
+      return currentMentorPreview;
+    });
+
+    const goals = waysDTO.map((wayDTO) => {
+      const goalPreview = goalsPreview
+        .find((elem) => elem.uuid === wayDTO.goalUuid);
+      if (!goalPreview) {
+        throw new Error(`${goalPreview} was not found`);
+      }
+
+      return goalPreview;
+    });
+
+    /**
+     * WayPreviewProps for each way separately
+     */
+    const getWayPreviewProps = (i: number) => {
+      const obj = {
+        owner: ownersPreview[i],
+        currentMentors: currentMentorsPreview[i],
+        goal: goals[i],
+      };
+
+      return obj;
+    };
+
+    const waysPreview = waysDTO
+      .map((wayDTO, i) => wayDTOToWayPreviewConverter(wayDTO, getWayPreviewProps(i)));
```

Comment:
This function should be very slow. It uses find inside map inside map. If we will have 100 users with 100 ways it means here we shuold have 1_000_000 operation just to render 100 items. it is really dangerous. Please at least leave comment

Reply:
Done